#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Gen Add (NAME, Datatype, Bitmask LEVEL, Description, Default Value, Min Value, Max Value)
gen.add("radius_limit", double_t, 0, "Search Radius Limit", 0.03, 0.01, 1.0)
gen.add("epsilon_angle", double_t, 0, "Descriptor Uncertainty Angle Bounds", 5.0, 0.0, 90.0)
gen.add("curvature_threshold", double_t, 0, "Curvature Threshold", 1.0, 0.01, 10.0)
gen.add("plane_radius", double_t, 0, "Plane Radius Tolerance", 0.1, 0.01, 1.0)
gen.add("size_weight", double_t, 0, "Weight dimensional parameters", 0.2, 0.0, 1.0)
gen.add("esf_weight", double_t, 0, "Weight ESF description", 0.2, 0.0, 1.0)
gen.add("lower_activation_threshold", double_t, 0, "Similar Object Found Threshold", 20.0, 0.0, 200.0)
gen.add("upper_activation_threshold", double_t, 0, "New Object Found Threshold", 75.0, 0.0, 200.0)
gen.add("object", int_t, 0, "Index for vector of classified objects", 0, 0, 100)


exit(gen.generate(PACKAGE, "perception", "Classify"))